internal/codegen/template.go

```diff
@@ -100,7 +99,9 @@ func (t *Template) Parse(_ *Stencil) error {
 // are rendered onto the Files field of the template struct.
 func (t *Template) Render(st *Stencil, vals *Values) error {
 	if len(t.Files) == 0 && !t.Library {
-		f, err := NewFile(strings.TrimSuffix(t.Path, ".tpl"), t.mode, t.modTime)
+		p := strings.TrimSuffix(t.Path, ".tpl")
+		p = t.Module.ApplyDirReplacements(p)
```

Comment:
**thought**: I wonder if it makes more sense to do this when the module returns the templates instead of at `Render` time.

Reply:
Thinking more, I'm pretty strongly thinking it should. If we keep it additive like this then we're going to have to handle this rendering logic other places, instead of getting it "for free" when you create a module.

Reply:
talking offline in discord -- you can't do this because you need the first render pass to complete before you can render the templates for the dirreplacements